packages/widgetbook_cli/bin/commands/publish.dart

```diff
@@ -0,0 +1,351 @@
+      )
+      ..addOption(
+        'base-commit',
+        help: 'The SHA hash of the commit of the base branch.',
+      )
+      ..addOption(
+        'github-token',
+        help: 'GitHub API token.',
+      )
+      ..addOption(
+        'pr',
+        help: 'The number of the PR.',
+      );
+  }
+
+  @override
+  final String description = 'Publish a new build';
+
+  @override
+  final String name = 'publish';
+
+  @override
+  Future<int> run() async {
+    final publishProgress = logger.progress(
+      'Uploading build',
+    );
+
+    final path = results['path'] as String;
+
+    if (!await GitDir.isGitDir(path)) {
+      publishProgress.fail();
+      usageException('Directory from "path" is not a Git folder');
+    }
+
+    final gitDir = await GitDir.fromExisting(
+      path,
+      allowSubdirectory: true,
+    );
+
+    publishProgress.update('Obtaining data from Git');
+
+    final isWorkingTreeClean = await gitDir.isWorkingTreeClean();
+
+    final apiKey = results['api-key'] as String;
+    final currentBranch = await gitDir.currentBranch();
+    final branch = results['branch'] as String? ?? currentBranch.branchName;
+
+    final commit = results['commit'] as String? ?? currentBranch.sha;
+
+    final gitProvider = results['git-provider'] as String;
+    final gitHubToken = results['github-token'] as String?;
+    final prNumber = results['pr'] as String?;
+
+    final baseBranch = results['base-branch'] as String?;
+
+    final ciArgsData = CliArgs(
+      apiKey: apiKey,
+      branch: branch,
+      commit: commit,
+      gitProvider: gitProvider,
+      gitHubToken: gitHubToken,
+      prNumber: prNumber,
+      baseBranch: baseBranch,
+      path: path,
+    );
+
+    final ciArgs = await CiParserRunner(
+      argResults: results,
+      gitDir: gitDir,
+    ).getParser()?.getCiArgs();
+
+    if (ciArgs == null) {
+      publishProgress.fail();
+      usageException(
+        'Your CI/CD pipeline provider is currently not supported.',
+      );
+    }
+
+    publishProgress.update('Checking commit');
+    if (!isWorkingTreeClean) {
+      logger.warn('You have un-commited changes');
+
+      final proceedWithUnCommitedChanges = logger.chooseOne(
+        'Would you like to proceed before you commit ?',
+        choices: ['no', 'yes'],
+        defaultValue: 'no',
+      );
+
+      switch (proceedWithUnCommitedChanges) {
+        case 'yes':
+          logger.warn('This might affect the expected results');
+          await _publishBuilds(
+            cliArgs: ciArgsData,
+            ciArgs: ciArgs,
+            gitDir: gitDir,
+            publishProgress: publishProgress,
+          );
+          break;
+
+        default:
+          publishProgress.cancel();
+          return ExitCode.success.code;
+      }
+    } else {
+      await _publishBuilds(
+        cliArgs: ciArgsData,
+        ciArgs: ciArgs,
+        gitDir: gitDir,
+        publishProgress: publishProgress,
+      );
```

Comment:
since we are also executing this method in line 185 to 189, we probably can improve the flow of the application by exiting if the user says no, otherwise just continuing with executing this method, deleting what is in line 185 to 189

Reply:
The first check is always on the highest level. when `isWorkingTree` is clean. (then ` _publishBuilds` is called, then the second invocation of `_publishBuilds` is when the user selects to continue with uploading with un committed changes